env0/resource_agent_project_assignment.go

```diff
@@ -57,138 +43,99 @@ func resourceAgentProjectAssignment() *schema.Resource {
 	}
 }
 
-func resourceAgentProjectAssignmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	var newAssignment AgentProjectAssignment
-	if err := readResourceData(&newAssignment, d); err != nil {
-		return diag.Errorf("schema resource data deserialization failed: %v", err)
-	}
-
-	apaLock.Lock()
-	defer apaLock.Unlock()
-
+func resourceAgentProjectAssignmentCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	apiClient := meta.(client.ApiClientInterface)
-	assignments, err := apiClient.ProjectsAgentsAssignments()
-	if err != nil {
-		return diag.Errorf("could not get project-agent assignments: %v", err)
-	}
 
-	for projectId, agentId := range assignments.ProjectsAgents {
-		if projectId == newAssignment.ProjectId && agentId == newAssignment.AgentId {
-			return diag.Errorf("assignment for project id %v and agent id %v already exist", projectId, agentId)
-		}
+	var assignment AgentProjectAssignment
+	if err := readResourceData(&assignment, d); err != nil {
+		return diag.Errorf("schema resource data deserialization failed: %v", err)
 	}
```

Comment:
We might still want this validation that was removed, but only making sure that there is no assignment, or that the assignment is to the default agent.

Removing this could mean that someone using a different Terraform/OpenTofu stack can attempt to manage the same resource there, and cause conflicts

Reply:
This is really what Terraform is for.. it will report a drift.
In any case, maintaining two states in Terraform is never good practice.

Reply:
Oh OK. So this error wasn't really necessary before, IIUC